libsel4muslcsys/src/vsyscall.c

```diff
@@ -292,3 +294,15 @@ long sel4_vsyscall(long sysnum, ...)
 /* Put a pointer to sel4_vsyscall in a special section so anyone loading us
  * knows how to configure our syscall table */
 uintptr_t VISIBLE SECTION("__vsyscall") __vsyscall_ptr = (uintptr_t) sel4_vsyscall;
+
+/* muslc provides a function used to initialise the C standard library
+ * environment. */
+extern void __init_libc(char const *const *envp, char const *pn);
+
+/* Initialise muslc environment */
+void CONSTRUCTOR(CONSTRUCTOR_MIN_PRIORITY) muslcsys_init_muslc(void)
```

Comment:
This constructor needs to happen before the `init_syscall_table` constructor, so it may need it's priority increased.

Reply:
I have 'raised' the priority of the other constructors so they will occur later as [lower-number priorities run first](https://gcc.gnu.org/onlinedocs/gcc-4.7.0/gcc/Function-Attributes.html).